prov/sockets/src/sock_av.c

```diff
@@ -452,6 +454,8 @@ static int sock_av_remove(struct fid_av *av, fi_addr_t *fi_addr, size_t count,
 				if (conn->av_index == idx)
 					conn->av_index = FI_ADDR_NOTAVAIL;
 				ofi_idm_clear(&sock_ep->attr->av_idm, idx);
+			} else {
+				ret = -FI_EINVAL;
```

Comment:
I don't think this is quite right. This is treating an existing conn that is in use as invalid. EINVAL should only be set if the conn doesn't exist ie `!conn`, not the second condition